### PR TITLE
mobile dashboard prototype landing page

### PR DESCRIPTION
**Pull request by czue**
Wanted to see what it would take to render a new version of the Program Summary as a skeleton / learning exercise for what it will take to fork off a set of mobile dashboard views and importing as little of all the dashboard cruft as possible.

This is what the page looks like in case curious:

image

**I recommend reviewing this all at once.**

Main changes:

- New mobile dashboard landing page
- New mobile dashboard angular app
- template provider service to swap template of a directive based on web/mobile status
- stub templates for program summary and KPIs to validate the approach.

### CODE REVIEW

File: custom/icds_reports/static/js/icds_app_mobile.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+var url = hqImport('hqwebapp/js/initial_page_data').reverse;
+
+function MainMobileController($scope, $route, $routeParams, $location, $window, $http,
```
Comment (czue) [inline custom/icds_reports/static/js/icds_app_mobile.js]: this is mostly copy/pasted and stripped down from `icds_app.js`

File: custom/icds_reports/static/js/angular-services/template-provider.service.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+window.angular.module('icdsApp').factory('templateProviderService', ['isMobile', function (isMobile) {
+    var url = hqImport('hqwebapp/js/initial_page_data').reverse;
+    return {
+        getTemplate: function (templateName) {
```
Comment (czue) [inline custom/icds_reports/static/js/angular-services/template-provider.service.js]: helper function to dynamically swap in a template based on a context variable controlling whether it's the mobile version or not

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/kpi.directive.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<div class="row" ng-repeat="section in $ctrl.data track by $index">
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/kpi.directive.html]: mostly copied from web version as a proof of concept.

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/program-summary.directive.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{% load i18n %}
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/program-summary.directive.html]: stripped down from the web version, as a proof of concept

File: custom/icds_reports/templates/icds_reports/mobile/dashboard/mobile_dashboard.html [hunk 5]
```diff
@@ -0,0 +1,88 @@
+{% load menu_tags %}
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/mobile/dashboard/mobile_dashboard.html]: copied and stripped down from the web version. I'm still unclear whether this should be a totally different page, whether they should extend the same base template, or whether they should be the same template with everything handled in context variables. but for now I wanted something standalone and separate so i could see the minimal set of stuff required to render a basic program summary.

**Review by czue — COMMENTED** [review]
@calellowitz @Rohit25negi @orangejenny I think this is ready for review.

File: custom/icds_reports/templates/icds_reports/mobile_login.html [hunk 6]
```diff
@@ -1,3 +1,14 @@
 {% extends "icds/login.html" %}
 {% load i18n %}
+{% load hq_shared_tags %}
 {% block title %}{% trans "Login to ICDS CAS Mobile Dashboard" %}{% endblock title %}
+{% block page_content %}
+{{ block.super }}
+{% registerurl 'cas_mobile_dashboard' domain %}
```
Comment (orangejenny) [inline custom/icds_reports/templates/icds_reports/mobile_login.html]: I think this can be removed; I don't see it referenced in js.

**Review by orangejenny — APPROVED** [review]
Didn't review that closely because I don't have much context on either the dashboard or angular, but that's probably fine since this is a prototype that copies a fair amount from the web dashboard.

File: custom/icds_reports/templates/icds_reports/mobile_login.html (hunk 6 above)
Comment (czue) [reply custom/icds_reports/templates/icds_reports/mobile_login.html]: Oops, I think I got my branches crossed. I fixed this in a rebase on https://github.com/dimagi/commcare-hq/pull/25950 but i didn't realize it was also on this branch...

File: custom/icds_reports/templates/icds_reports/mobile_login.html [hunk 7]
```diff
@@ -1,3 +1,14 @@
...
+<script type="text/javascript">
```
[the remark this answers is not in the document]
Comment (czue) [reply custom/icds_reports/templates/icds_reports/mobile_login.html]: Interesting. Is it considered best practice to not include it?

File: custom/icds_reports/templates/icds_reports/mobile_login.html (hunk 6 above)
Comment (czue) [reply custom/icds_reports/templates/icds_reports/mobile_login.html]: Actually nevermind I didn't actually fix it there either. Will fix now.

File: custom/icds_reports/templates/icds_reports/mobile_login.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (orangejenny) [reply custom/icds_reports/templates/icds_reports/mobile_login.html]: In HQ's codebase, yes, for consistency's sake don't include it.

https://github.com/dimagi/commcare-hq/pull/12632

[the remark this answers is not in the document]
Comment (czue) [reply custom/icds_reports/templates/icds_reports/mobile_login.html]: Good to know thanks

**Pull request merged**